libvcx/src/aries/handlers/proof_presentation/prover/state_machine.rs

```diff
@@ -123,15 +126,11 @@ impl ProverSM {
                         }
                     }
                     ProverMessages::RejectPresentationRequest((reason)) => {
-                        let connection_handle = connection_handle
-                            .ok_or(VcxError::from_msg(VcxErrorKind::ActionNotSupported, "Presentation is already sent"))?;
-                        Self::_handle_reject_presentation_request(connection_handle, &reason, &state.presentation_request, &thread_id)?;
+                        Self::_handle_reject_presentation_request(send_message.unwrap(), &reason, &state.presentation_request, &thread_id)?;
```

Comment:
I am very uncomfortable with these unwraps - I think `_handle_reject_presentation_request` and `_handle_presentation_proposal` should take `send_message` as `Option` and return appropriate error if it is `None`.